zcash_client_sqlite/src/prepared.rs

```diff
@@ -478,7 +482,8 @@ impl<'a, P> DataConnStmtCache<'a, P> {
             u32::from(account),
             to_str,
             ivalue,
-            memo.map(|m| m.as_slice()),
+            memo.filter(|m| *m != &MemoBytes::empty())
+                .map(|m| m.as_slice()),
```

Comment:
I think it would be a really bad thing if downstream consumers used that as a sentinel for download state, which leads me to this resolution: https://github.com/zcash/librustzcash/pull/620/commits/3ca50520d440f6111277a2f6d8b0ea2088afbf18 - uniformly avoid storing the empty memo; this also should avoid a potentially-costly memcmp in the new views. This might be a premature optimization, but I would much prefer that callers not be able to use the nullity of the memo field for determining download state; if we want that information to be available, it should be via an explicit status flag.

Reply:
My concern was that downstream consumers might _already_ be using the nullity of the memo field for this purpose. But I believe it's more likely that they are using nullity of the `raw` field of the corresponding entry in the `transactions` table for this purpose, so this change _should be fine.